### PR TITLE
GH-111485: Allow arbitrary annotations on instructions and micro-ops.

### PR DESCRIPTION
**Pull request by markshannon**
This allows us to be explicit about classifications of uops that are currently encoded in the code generator or analysis phase.

As an example, I've added a "specializing" annotation to denote which micro-ops specialize, rather than perform an action.
Using that we can avoid emitting the specialization code in tier 2.

A possible extension is to allow more than one annotation, but this PR only allows one for now.

* Issue: gh-111485

### CODE REVIEW

**Review by gvanrossum — COMMENTED** [review]
Nice work, but I think the Meta folks are going to be upset that you are breaking "override". They use that feature. @itamaro @carljm 

Also, `test_generated_cases` is now broken. One possible fix would be to add `"annotation"` to the set `annotations` in lexer.py.

**Comment by gvanrossum** [comment]
Maybe we need to add a test for `override` to `test_generated_cases.py`, so we know it doesn't rot.

**Comment by jbower-fb** [comment]
> Nice work, but I think the Meta folks are going to be upset that you are breaking "override". They use that feature. @itamaro @carljm

Yes, we definitely still want this please.

> Maybe we need to add a test for override to test_generated_cases.py, so we know it doesn't rot.

There were some tests, but this PR removed them.

@markshannon is there some reason `override` can't co-exist with annotations?

**Comment by markshannon** [comment]
Since it is not clear to me if and when `override` is used, I've change the code to allow multiple annotations per op/inst.

**Comment by jbower-fb** [comment]
Great, thank you for doing that.

The motivation for `override` is to make it a bit easier/safer for things like Cinder where we need to make our own tweaks to the interpreter. See https://github.com/python/cpython/issues/102021 where this was introduced.

File: Tools/cases_generator/parsing.py [hunk 1]
```diff
@@ -174,21 +171,23 @@ def inst_def(self) -> InstDef | None:
...
     def inst_header(self) -> InstHeader | None:
-        # [override] inst(NAME)
-        #   | [override] [register] inst(NAME, (inputs -- outputs))
-        #   | [override] [register] op(NAME, (inputs -- outputs))
-        # TODO: Make INST a keyword in the lexer.
-        override = bool(self.expect(lx.OVERRIDE))
-        register = bool(self.expect(lx.REGISTER))
-        if (tkn := self.expect(lx.IDENTIFIER)) and tkn.text in ("inst", "op"):
+        # [annotation] inst(NAME, (inputs -- outputs))
+        # | [annotation] op(NAME, (inputs -- outputs))
```
Comment (gvanrossum) [inline Tools/cases_generator/parsing.py]:
```suggestion
        # annotation* inst(NAME, (inputs -- outputs))
        # | annotation* op(NAME, (inputs -- outputs))
```

**Review by gvanrossum — APPROVED** [review]
Awesome!

**Pull request merged**